test/e2e/index.test.ts

```diff
@@ -338,5 +338,17 @@ describe('e2e', () => {
       });
       expect(scrollLeft).not.to.equal(0);
     });
+
+    // https://github.com/mui/mui-x/issues/4190
+    it('should move the focus from left pinned column to the cell in main render zone after pressing Tab during row editing', async () => {
```

Comment:
I tried to add a unit test for it, but wasn't able to make `Tab` key work, hence e2e test